### DISCUSSION TITLE
Error: could not find function "file.edit" Execution halted  Unable to locate an installed version of R.

### DISCUSSION THREAD

**Discussion by brendanmagill** [quarto-dev/quarto-cli#3552] [Q&A] [answered]
I am able to run R code within a Quarto document, but when I try to render the document (either html or pdf) I receive the following error:

Error: could not find function "file.edit"
Execution halted
Error: could not find function "file.edit"
Execution halted

Unable to locate an installed version of R.
Install R from https://cloud.r-project.org/

Running 'quarto check' gives the same error: 'Error: could not find function "file.edit"' which results in not finding an installed version of R. I have tried uninstalling both RStudio and R, and reinstalling both. Any ideas?

**Comment by mcanouil** [comment]
Could you provide a small reproducible example? Without nobody will be able to help.
Also could you give some context? What operating system? What Quarto version? etc.

**Reply by brendanmagill** [reply]
I am using Windows 11 with Quarto version 1.2. I am using Quarto within RStudio. When I render a file without R code, it works fine. But any qmd file containing R code gives this error. For example, I get this error when rendering the following snippet.
<img width="196" alt="image" src="https://user-images.githubusercontent.com/66447760/208567216-201f5a8e-4bca-4923-bbe2-d6a401f237d0.png">

**Reply by mcanouil** [reply]
DO you have the same error while running Quarto CLI in PowerShell for example?

**Reply by brendanmagill** [reply]
The computer with this issue accidentally got melted on a stovetop - guess we will never know what the issue was!

**Comment by MarcoJCTeng** [comment]
I am also facing the same problem as you, but with the MacOS M1 chip. I even tried rendering with the whole new qmd, but the result was disappointing. 

<img width="772" alt="Screenshot 2023-04-29 at 12 50 24 AM" src="https://user-images.githubusercontent.com/123462382/235206966-e0614b85-bc52-4292-8a08-a159b081deea.png">

Reinstallation was one of my solutions, but it proved to be useless. Afterward, I tried to knit it with rmd, and  I got the same result as qmd did. 

<img width="775" alt="Screenshot 2023-04-29 at 1 03 02 AM" src="https://user-images.githubusercontent.com/123462382/235209370-524e219a-ecc8-4884-946f-dece2a444945.png">

It would be greatly appreciated if anyone could help me solve the problem.

**Reply by cscheid** [reply]
If you're experiencing the issue with knit and rmd, then it can't be a quarto problem :), since you're using rmarkdown.

**Reply by MarcoJCTeng** [reply]
Sorry for not making my statement clear. The first figure that I posted is with qmd. the rendering function seems not working well here, and "Error: could not find function "file.edit" Execution halted" was what I received. Therefore, I could not even further publish on RPubs or any other platform. 

The second figure, which was created using R Markdown, is being shared as an alternative option with others.

**Reply by mcanouil** [reply]
The issue is, even from the original post, we are unable to reproduce what seems to be an edge case.

Could you provide a fully reproducible example and the steps required to reproduce this?
This means to also describe the version of the tools you are using.

See details on how to post "bug reports" in <https://quarto.org/bug-reports.html>.

Also, did you try to render using Quarto CLI inside the integrated terminal of RStudio IDE? If not, could you?

**Reply by cderv** [reply]
It seems to me this is an environment issue. I don't think this is related to Quarto said. 

Possibly something related to RStudio which calls `file.edit()` (from **utils** 📦 ) in its code. To check this: 

* Install Quarto https://quarto.org/docs/download/
* Run in terminal `quarto check` to see if R is correctly found
* Render your document outside of RStudio using in terminal `quarto render`

Do you see still see the issue ? If not, then this is a RStudio issue. So next: 

* Do you use latest RStudio IDE ? 
* If so, open an issue in RStudio repo. 
Hope it helps

**Reply by MarcoJCTeng** [reply]
Hi, thank you for replying. I also think that should be some problems within the packages `utils` in its code. 

<img width="1193" alt="Screenshot 2023-05-04 at 2 43 27 PM" src="https://user-images.githubusercontent.com/123462382/236129610-15a0e66d-83e2-45a7-9057-301ed896c11a.png">

When I tried to install the `utils` package, this warning kept popping up even when I clicked 'Yes'.

<img width="911" alt="Screenshot 2023-05-04 at 2 44 37 PM" src="https://user-images.githubusercontent.com/123462382/236130013-8f03e6b2-c0b3-4ee1-a3ba-aac7a4cc93bd.png">

Back to the suggestion, I've followed the instruction and run `quarto check`  in terminal. You can see this by following the image.

<img width="780" alt="Screenshot 2023-05-04 at 2 21 27 PM" src="https://user-images.githubusercontent.com/123462382/236125991-5bbe5280-585a-4410-91e0-1e881fa9ea05.png">

**Reply by cderv** [reply]
> I've followed the instruction and run quarto check in terminal. You can see this by following the image.

I meant a terminal outside of RStudio IDE. 

1. Install Quarto [quarto.org/docs/download](https://quarto.org/docs/download/)
1. Run in terminal from OS **outside of rstudio** `quarto check` to see if R is correctly found
1. Render your document outside of RStudio using in terminal quarto render

> be some problems within the packages utils in its code.

Before trying to install, can you check that **utils** package is already installed ? `packageVersion("utils")` ? 
It should be install with R. 

What is the RStudio IDE version you are using ?

**Comment by MarcoJCTeng** [comment, marked as answer]
Sorry for the super late reply. I have solved the problem by reinstalling my MacBook. Once again, thank you for your help in solving the problems, even though we still do not know what really happened.
